ICSharpCode.CodeConverter/CSharp/NodesVisitor.cs

```diff
@@ -1162,11 +1162,13 @@ public override CSharpSyntaxNode VisitArrayCreationExpression(VBSyntax.ArrayCrea
 
             public override CSharpSyntaxNode VisitCollectionInitializer(VBSyntax.CollectionInitializerSyntax node)
             {
-                var typeInfo = _semanticModel.GetTypeInfo(node);
-                var isArrayType = typeInfo.ConvertedType?.IsKind(SymbolKind.ArrayType) == true;
-                var initializerType = isArrayType ? SyntaxKind.ArrayInitializerExpression : SyntaxKind.CollectionInitializerExpression;
+                var isArrayLiteral =
+                    !( node.Parent is VBSyntax.ObjectCollectionInitializerSyntax
+                    || node.Parent is VBSyntax.CollectionInitializerSyntax
+                    || node.Parent is VBSyntax.ArrayCreationExpressionSyntax);
```

Comment:
Looks good! I just inverted this logic to simplify it slightly in https://github.com/icsharpcode/CodeConverter/commit/38dda0bf98d22e24181e2c2df8e7f51b13b24f3b